### PR TITLE
Add NEMO4 DOMAINcfg tool callgraph #2

### PR DESCRIPTION
**Pull request by oceandie**
@pyNEMO/pydomcfg I've added NEMO4 DOMAINcfg tool callgraph doxygen documentation to our documentation. The structure is not definitive at all - I just tried that the solution can work, then we can decide together how to make it nicer/clearer.
#2

### CODE REVIEW

**Comment by oceandie** [comment]
@pyNEMO/pydomcfg Do we want to introduce the structure of pyDOMCFG code in this PR? if yes, how would you recommend we do it?

**Comment by jdha** [comment]
> @pyNEMO/pydomcfg Do we want to introduce the structure of pyDOMCFG code in this PR? if yes, how would you recommend we do it?

Happy for this PR to be merged. Perhaps if you open a new Issue and propose a structure, Mattia and I can then comment and evolve the structure to something we're all agreed on before opening a branch. Perhaps easier than all trying to come up with something at the same time.

File: docs/readme.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+********
+Overview
+********
+
+pyDOMCFG bla bla bla `NEMO  <https://www.nemo-ocean.eu/>`_, bla bla bla `NEMO4 DOMAINcfg tool  <https://forge.ipsl.jussieu.fr/nemo/chrome/site/doc/NEMO/guide/html/tools.html>`_.
```
Comment (malmans2) [inline docs/readme.rst:5]: https://www.youtube.com/watch?v=Hrph2EW9VjY

**Review by malmans2 — APPROVED** [review]

**Comment by malmans2** [comment]
> > @pyNEMO/pydomcfg Do we want to introduce the structure of pyDOMCFG code in this PR? if yes, how would you recommend we do it?
> 
> Happy for this PR to be merged. Perhaps if you open a new Issue and propose a structure, Mattia and I can then comment and evolve the structure to something we're all agreed on before opening a branch. Perhaps easier than all trying to come up with something at the same time.

Sounds good!

**Comment by oceandie** [comment]
> 
> 
> > @pyNEMO/pydomcfg Do we want to introduce the structure of pyDOMCFG code in this PR? if yes, how would you recommend we do it?
> 
> Happy for this PR to be merged. Perhaps if you open a new Issue and propose a structure, Mattia and I can then comment and evolve the structure to something we're all agreed on before opening a branch. Perhaps easier than all trying to come up with something at the same time.

Sounds good to me too!

**Pull request merged**